### PR TITLE
Pvl Core Addition

### PR DESCRIPTION
**Pull request by acpaquette**
## Description
This PR adds a long standing effort to extract the core file reading and writing components from the main `src` files in ISIS into there own, individually buildable library. All files that were moved should match the files in dev and should just be moved into a new folder.

## Related Issue
N/A

## How Has This Been Validated?
Validated through passing Jenkins tests

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation change (update to the documentation; no code change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Infrastructure change (changes to things like CI or the build system that do not impact users)

## Checklist:

- [x] I have read and agree to abide by the [Code of Conduct](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/Code-Of-Conduct.md)
- [x] I have read the [**CONTRIBUTING**](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CONTRIBUTING.md) document.
- [ ] My change requires a change to the documentation and I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] I have added myself to the [.zenodo.json](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/.zenodo.json) document.
- [x] I have added any user impacting changes to the [CHANGELOG.md](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CHANGELOG.md) document.

## Licensing
This project is mostly composed of free and unencumbered software released into the public domain, and we are unlikely to accept contributions that are not also released into the public domain. Somewhere near the top of each file should have these words:

> This work is free and unencumbered software released into the public domain. In jurisdictions that recognize copyright laws, the author or authors of this software dedicate any and all copyright interest in the software to the public domain.

- [x] I dedicate any and all copyright interest in this software to the public domain. I make this dedication for the benefit of the public at large and to the detriment of my heirs and successors. I intend this dedication to be an overt act of relinquishment in perpetuity of all present and future rights to this software under copyright law.

### CODE REVIEW

File: isis/src/base/objs/Application/Application.cpp [hunk 1]
```diff
@@ -387,6 +382,21 @@ namespace Isis {
...
+   * @param results Pvl containing the results to add to the session log
+   * @param log App log for running in code
+   *
+   */
+  void Application::AppendAndLog(PvlGroup &results, Pvl *log) {
+    Application::Log(results);
+    if (log) {
+      log->addGroup(results);
+    }
+  }
```
Comment (Kelvinrr) [inline isis/src/base/objs/Application/Application.cpp:398]: Oof, I forgot about this one. Would this constitute an API breaking change? 🤔

File: isis/src/core/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,113 @@
+cmake_minimum_required(VERSION 3.10)
+
+project (core VERSION 0.0.1)
```
Comment (jlaura) [inline isis/src/core/CMakeLists.txt]: Is PVL really a 0.0.1 version? That would be like a just developed, alpha, probably do not use this because nothing is stable version. What are the criteria for this to be a 1.0.0 release that folks should be comfortable using?

File: rtd_docs/conf.py [hunk 3]
```diff
@@ -49,4 +49,18 @@
...
+
+if read_the_docs_build:
+    subprocess.call('doxygen Doxyfile', shell=True)
+
+# -- Breathe Configuration ----------------------------------------------------
+
+breathe_projects = { "ISIS": "doxyxml/" }
+breathe_default_project = "ISIS"
+
+breathe_domain_by_extension = {"h" : "cpp"}
```
Comment (Kelvinrr) [inline rtd_docs/conf.py:66]: Is this related to the core stuff? Seems like an odd inclusion. Im guessing it's for building the PVL core with docs. If so we might want to add a new objective of getting the docs pushed to the S3 buckets and the landing page, but that can happen after this is merged.

**Comment by amystamile-usgs** [comment]
I'm sure you are already aware of these test failures:

	2077 - Mosrange.MosrangeDefault (Failed)
	2078 - Mosrange.MosrangeOnErrorContinue (Failed)
	2079 - Mosrange.MosrangeOnErrorFail (Failed)

File: isis/src/core/CMakeLists.txt (hunk 2 above)
Comment (Kelvinrr) [reply isis/src/core/CMakeLists.txt]: I think we agreed a while back that if it's going public it should be 1.0, So I imagine this should also be 1.0? We plan to do a conda deploy after this is merged.

**Comment by Kelvinrr** [comment]
> I'm sure you are already aware of these test failures:
> 
> ```
> 2077 - Mosrange.MosrangeDefault (Failed)
> 2078 - Mosrange.MosrangeOnErrorContinue (Failed)
> 2079 - Mosrange.MosrangeOnErrorFail (Failed)
> ```

These may be hold overs from a previous PR that fixed the errors. We probably need updated tests.

File: isis/src/core/CMakeLists.txt (hunk 2 above)
Comment (acpaquette) [reply isis/src/core/CMakeLists.txt]: Agree with both, I just used 0.0.1 as a place holder

File: isis/src/base/objs/Application/Application.cpp (hunk 1 above)
Comment (acpaquette) [reply isis/src/base/objs/Application/Application.cpp:398]: I don't think so as I didn't remove any functionality but it is an addition. Likely an 8.X.0 release

File: isis/src/core/CMakeLists.txt (hunk 2 above)
Comment (acpaquette) [reply isis/src/core/CMakeLists.txt]: Updated to 1.0.0

**Comment by Kelvinrr** [comment]
hmmm, those tests are still failing so maybe they aren't from the previous PR.

**Comment by KrisBecker** [comment]
> hmmm, those tests are still failing so maybe they aren't from the previous PR.

The `pvl_core` branch is one commit behind `dev`. That commit fixes these test failures. You need to merge `dev` into the `pvl_core` branch to fix it.

**Comment by acpaquette** [comment]
@KrisBecker I rebased off of dev and the issue was still present. There is an extra function to both log to an open application window and add to the pvl log output. I have fixed the issue and this should be good to merge

**Comment by acpaquette** [comment]
@KrisBecker To address some of the API breaking issues. I had to move the logging from `PvlGroup` which accessed the `Application` class. This was not an ideal coupling. I changed this so that the `Application` class would log to both the application and the resulting pvl output to maintain current functionality. 

The resulting test failures were from a poor git rebase on my part. Regardless, there is an API breakage under the `PvlObject` where the `appendLogGroup` function has been stripped of any functionality. This was done after the decouple. Unfortunately with the semantic versioning, this PR will have to wait a year for that function to to deprecated and the functionality to be added under 9.0. I will close this PR and start working on getting the application log changes into its own PR.
